Parse/src/main/java/com/parse/ParseHttpResponse.java

```diff
@@ -50,7 +54,12 @@ public T setReasonPhase(String reasonPhase) {
     }
 
     public T setHeaders(Map<String, String> headers) {
-      this.headers = Collections.unmodifiableMap(new HashMap<>(headers));
+      this.headers = new HashMap<>(headers);
+      return self();
+    }
```

Comment:
Should we change this to:

```java
public T addHeaders(Map<String, String> headers) {
  this.headers.addAll(headers);
  return self();
}
```

Leaving it as `setHeaders` where it can clear all old values seem like it could be confusing.